reader/src/main/java/org/kaazing/monitoring/reader/api/MMFReader.java

```diff
@@ -19,37 +19,44 @@
  * under the License.
  */
 
-package org.kaazing.monitoring.reader;
+package org.kaazing.monitoring.reader.api;
 
-import java.io.File;
-import java.nio.MappedByteBuffer;
-
-import org.kaazing.monitoring.reader.file.location.MonitoringFolderAgrona;
-import org.kaazing.monitoring.reader.impl.file.location.MonitoringFolderAgronaImpl;
-
-import uk.co.real_logic.agrona.IoUtil;
+import java.util.List;
 
 /**
- * Processes the Agrona monitoring file
- *
+ * This interface is used for reading data from the MMF
  */
-public class FileProcessor {
+public interface MMFReader {
+
+    /**
+     * Returns the gateway id
+     * @return String
+     */
+    String getGatewayId();
 
-    private String fileName;
-    private static MonitoringFolderAgrona agronaFolder = new MonitoringFolderAgronaImpl();
+    /**
+     * Returns the MMF's version
+     * @return int
+     */
+    int getFileVersion();
 
-    public FileProcessor(String fileName) {
-        this.fileName = fileName;
-    }
+    /**
+     * Returns a list with all the collected gateway counters
+     * @return List<Counter>
+     */
+    List<Counter> getGatewayCounters();
 
     /**
-     * Initializes and maps the Agrona monitoring file
-     * @return MappedByteBuffer - the Agrona mapped file
+     * Returns a list with all the existing services
+     * @return List<Service>
      */
-    public MappedByteBuffer getMappedFile() {
-        File file = new File(agronaFolder.getMonitoringDir(), fileName);
+    List<ServiceCounters> getServices();
 
-        return IoUtil.mapExistingFile(file, fileName);
-    }
+    /**
+     * Returns a list with all the collected service counters for a given service
+     * @param service
+     * @return List<Counter>
+     */
+    List<Counter> getServiceCounters(ServiceCounters service);
```

Comment:
This method seems to overlap a bit with getServices(). And the name implies it should return a ServiceCounters instance. I would suggest either removing this method or changing it to take the service name as the input parameter and return ServiceCounters:
```
ServiceCounter getServiceCounters(String serviceName)
```